Add colored edges to the cards

fkz opened the pull request:
Based on the type, add colored edges. The colors are picked somewhat randomly for now:
- yellow for actions
- blue for items
- green for spells

Test print:

![20230828_012622](https://github.com/potsdam-pnp/pathfinder2-cards/assets/961451/998dece8-3383-4f1e-bf3c-5cc2df64da2d)

fkz committed:
Add colored edges to the cards

Based on the type, add colored edges. The colors are picked somewhat randomly for now:
- yellow for actions
- blue for items
- green for spells

We take care to do the following:
 - Cardcolor is set to 40%
 - Rounded corners are used for card frames
 - Use @ in names of internal macros

Toxaris commented on card-style.cls:
```diff
@@ -36,6 +37,27 @@
+}
+
+\xdef\CurrentCardCategory{None}
+\AddToHook {shipout/background}{\CardType{\CurrentCardCategory}}
```
I'm confused about the words "category" vs "type" here.

fkz replied:
Indeed, we should use more consistent phrasing.

Toxaris commented on card-style.cls:
```diff
@@ -36,6 +37,27 @@
+  \ifnum\pdfstrcmp{#1}{Cantrip}=0   \def\cardcolor{green} \fi
+  \ifnum\pdfstrcmp{#1}{Focus}=0     \def\cardcolor{green} \fi
+  
+  \ifnum\pdfstrcmp{\cardcolor}{}=0 \else
+    \tikz[remember picture,overlay] {%
+      \draw [color=\cardcolor,line width=2mm]
+        (current page.south west)
+        rectangle
+        (current page.north east);
+    }
+  \fi
```
round corners?

fkz replied:
Yes, I was still in the phase of getting something working at all 😃

Toxaris commented on card-style.cls:
```diff
@@ -36,6 +37,27 @@
 
+\newcommand{\CardType}[1]{%
+  \def\cardcolor{}
+  \ifnum\pdfstrcmp{#1}{Item}=0      \def\cardcolor{blue} \fi
+  \ifnum\pdfstrcmp{#1}{Action}=0    \def\cardcolor{yellow} \fi
+  \ifnum\pdfstrcmp{#1}{Spell}=0     \def\cardcolor{green} \fi
+  \ifnum\pdfstrcmp{#1}{Cantrip}=0   \def\cardcolor{green} \fi
+  \ifnum\pdfstrcmp{#1}{Focus}=0     \def\cardcolor{green} \fi
```
This looks like something `pgfkeys` can do for us, similar to `every ...` in tikz. Along these lines:

```latex
\tikzset{card background=line width=2mm}
\tikzset{Item card background=card background, color=blue}
...

\newcommand{\CardType}[1]{%
  \tikz[remember picture, overlay]{%
    \draw[#1 card background]
      ...
```

fkz replied:
Oh, yes that looks nice

Toxaris commented on card-style.cls:
```diff
@@ -36,6 +37,27 @@
 
+\newcommand{\CardType}[1]{%
+  \def\cardcolor{}
+  \ifnum\pdfstrcmp{#1}{Item}=0      \def\cardcolor{blue} \fi
```
somehow I was assuming we distinguish between permanent and consumable items

fkz replied:
🤷

Toxaris reviewed (commented):
Some thoughts.

fkz committed:
Fix detect-magic: Don't use variable named \lower

Apparantly that conflicts with something else overriding it,
and that leads to compiel errors.

It works again after renaming it accordingly.

fkz commented:
@Toxaris Feel free to work further on this, I'll stop working on it for now.

fkz commented:
Ah, ich war die ganze Zeit verwirrt, warum hier der latex-Check nicht kommt, aber das liegt daran, dass der PR von nem Fork ausgeht und deswegen nicht die Berechtigung dazu hat. Und die Fehlerbehandlung ist dann so, dass alles grün ist.
